### PR TITLE
Added function for US4 - Add Transfer Method - Send event when accoun…

### PR DESCRIPTION
**Pull request by sarun-altmtrk**
…t information collection screen is shown.

### CODE REVIEW

File: TransferMethod/Tests/AddTransferMethodPresenterTests.swift [hunk 1]
```diff
@@ -8,6 +8,8 @@ class AddTransferMethodPresenterTests: XCTestCase {
     private let mockView = MockAddTransferMethodViewTests()
     private lazy var transferMethodConfigurationFieldsResponse = HyperwalletTestHelper
         .getDataFromJson("TransferMethodConfigurationFieldsResponse")
+    private lazy var trackTransferMethodConfigurationFieldsResponse = HyperwalletTestHelper
+    .getDataFromJson("TrackTransfermethodConfigurationFieldsResponse")
```
[the remark this answers is not in the document]
Comment (sarun-altmtrk) [reply TransferMethod/Tests/AddTransferMethodPresenterTests.swift]: Fixed

File: TransferMethod/Tests/AddTransferMethodPresenterTests.swift [hunk 2]
```diff
@@ -265,6 +267,13 @@ class AddTransferMethodPresenterTests: XCTestCase {
         return HyperwalletTestHelper.buildPostRequest(baseUrl: HyperwalletTestHelper.graphQlURL, response)
     }
 
+    private func setupTrackTransferMethodConfigurationFields(_ error: NSError? = nil) -> StubRequest {
```
[the remark this answers is not in the document]
Comment (sarun-altmtrk) [reply TransferMethod/Tests/AddTransferMethodPresenterTests.swift]: Fixed

File: TransferMethod/Tests/Helper/HyperwalletTestHelper.swift [hunk 3]
```diff
@@ -9,6 +9,7 @@ class HyperwalletTestHelper {
     static let contentType = "Content-Type"
     static let graphQlURL = "https://localhost/graphql"
     static let restURL = "https://localhost/rest/v3/"
+    static let insightsUrl = "https://localhost/insight"
```
Comment (jsaini-hw) [inline TransferMethod/Tests/Helper/HyperwalletTestHelper.swift]: why do you need this?

File: TransferMethod/Tests/AddTransferMethodPresenterTests.swift [hunk 4]
```diff
@@ -8,6 +8,8 @@ class AddTransferMethodPresenterTests: XCTestCase {
     private let mockView = MockAddTransferMethodViewTests()
     private lazy var transferMethodConfigurationFieldsResponse = HyperwalletTestHelper
         .getDataFromJson("TransferMethodConfigurationFieldsResponse")
+    private lazy var trackTransferMethodConfigurationFieldsResponse = HyperwalletTestHelper
+    .getDataFromJson("TrackTransferMethodConfigurationFieldsResponse")
```
Comment (jsaini-hw) [inline TransferMethod/Tests/AddTransferMethodPresenterTests.swift]: why do you need this?

File: TransferMethod/Tests/InsightsDBHelper.swift [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import CoreData
+import Foundation
+@testable import Insights
+import XCTest
+
+class InsightsDBHelper: XCTestCase {
+    var mgr: EventManager?
```
Comment (jsaini-hw) [inline TransferMethod/Tests/InsightsDBHelper.swift]: rename mgr -> manager

File: TransferMethod/Tests/Responses/TrackTransfermethodConfigurationFieldsResponse.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (jsaini-hw) [inline TransferMethod/Tests/Responses/TrackTransfermethodConfigurationFieldsResponse.json]: why do you need this?

File: TransferMethod/Tests/Helper/HyperwalletTestHelper.swift (hunk 3 above)
Comment (sarun-altmtrk) [reply TransferMethod/Tests/Helper/HyperwalletTestHelper.swift]: Removed it

File: TransferMethod/Tests/AddTransferMethodPresenterTests.swift (hunk 4 above)
Comment (sarun-altmtrk) [reply TransferMethod/Tests/AddTransferMethodPresenterTests.swift]: Removed it

File: TransferMethod/Tests/InsightsDBHelper.swift (hunk 5 above)
Comment (sarun-altmtrk) [reply TransferMethod/Tests/InsightsDBHelper.swift]: Updated

File: TransferMethod/Tests/Responses/TrackTransfermethodConfigurationFieldsResponse.json (hunk 6 above)
Comment (sarun-altmtrk) [reply TransferMethod/Tests/Responses/TrackTransfermethodConfigurationFieldsResponse.json]: Removed the file

File: TransferMethod/Sources/AddTransferMethodPresenter.swift [hunk 7]
```diff
@@ -232,4 +236,15 @@ final class AddTransferMethodPresenter {
     private func resetErrorMessagesForAllSections() {
         sectionData.forEach { $0.errorMessage = nil }
     }
+
+    private func trackUILoadImpression () {
+        HyperwalletInsights.shared.trackImpression(pageName: pageName,
+                                                   pageGroup: pageGroup,
+                                                   params:
+            [
+                InsightsTags.country: country, InsightsTags.currency: currency,
```
Comment (ddhar-hw) [inline TransferMethod/Sources/AddTransferMethodPresenter.swift]: Please format this in different lines

File: HyperwalletUISDK.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -4413,7 +4429,10 @@
 				DYLIB_COMPATIBILITY_VERSION = 1;
 				DYLIB_CURRENT_VERSION = 1;
 				DYLIB_INSTALL_NAME_BASE = "@rpath";
-				FRAMEWORK_SEARCH_PATHS = "$(PROJECT_DIR)/Carthage/Build/iOS";
+				FRAMEWORK_SEARCH_PATHS = (
+					"$(PROJECT_DIR)/Carthage/Build/iOS",
+					"$(PROJECT_DIR)",
```
Comment (jsaini-hw) [inline HyperwalletUISDK.xcodeproj/project.pbxproj]: why this change?

File: HyperwalletUISDK.xcodeproj/project.pbxproj [hunk 9]
```diff
@@ -4386,7 +4399,10 @@
...
```
Comment (jsaini-hw) [inline HyperwalletUISDK.xcodeproj/project.pbxproj]: why this change?

File: HyperwalletUISDK.xcodeproj/project.pbxproj [hunk 10]
```diff
@@ -4156,6 +4164,7 @@
 				FRAMEWORK_SEARCH_PATHS = (
 					"$(inherited)",
 					"$(PROJECT_DIR)/Carthage/Build/iOS",
+					"$(PROJECT_DIR)",
```
Comment (jsaini-hw) [inline HyperwalletUISDK.xcodeproj/project.pbxproj]: why this change?

File: HyperwalletUISDK.xcodeproj/project.pbxproj [hunk 11]
```diff
@@ -4185,6 +4194,7 @@
...
```
Comment (jsaini-hw) [inline HyperwalletUISDK.xcodeproj/project.pbxproj]: why this changes?

File: HyperwalletUISDK.xcodeproj/project.pbxproj [hunk 12]
```diff
@@ -4359,7 +4369,10 @@
 				DYLIB_COMPATIBILITY_VERSION = 1;
 				DYLIB_CURRENT_VERSION = 1;
 				DYLIB_INSTALL_NAME_BASE = "@rpath";
-				FRAMEWORK_SEARCH_PATHS = "$(PROJECT_DIR)/Carthage/Build/iOS";
+				FRAMEWORK_SEARCH_PATHS = (
+					"$(PROJECT_DIR)/Carthage/Build/iOS",
+					"$(PROJECT_DIR)",
```
Comment (jsaini-hw) [inline HyperwalletUISDK.xcodeproj/project.pbxproj]: why this change? remove all of these changes

File: HyperwalletUISDK.xcodeproj/project.pbxproj [hunk 13]
```diff
@@ -4436,7 +4455,10 @@
 				ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES = YES;
 				CODE_SIGN_IDENTITY = "iPhone Developer";
 				CODE_SIGN_STYLE = Automatic;
-				FRAMEWORK_SEARCH_PATHS = "$(PROJECT_DIR)/Carthage/Build/iOS";
+				FRAMEWORK_SEARCH_PATHS = (
+					"$(PROJECT_DIR)/Carthage/Build/iOS",
+					"$(PROJECT_DIR)",
```
Comment (jsaini-hw) [inline HyperwalletUISDK.xcodeproj/project.pbxproj]: remove this

File: HyperwalletUISDK.xcodeproj/project.pbxproj [hunk 14]
```diff
@@ -4458,7 +4480,10 @@
...
```
Comment (jsaini-hw) [inline HyperwalletUISDK.xcodeproj/project.pbxproj]: remove this

File: HyperwalletUISDK.xcodeproj/project.pbxproj [hunk 15]
```diff
@@ -4480,7 +4505,10 @@
...
```
Comment (jsaini-hw) [inline HyperwalletUISDK.xcodeproj/project.pbxproj]: remove this

File: HyperwalletUISDK.xcodeproj/project.pbxproj [hunk 16]
```diff
@@ -4599,6 +4627,7 @@
 				FRAMEWORK_SEARCH_PATHS = (
 					"$(inherited)",
 					"$(PROJECT_DIR)/Carthage/Build/iOS",
+					"$(PROJECT_DIR)",
```
Comment (jsaini-hw) [inline HyperwalletUISDK.xcodeproj/project.pbxproj]: remove this

File: TransferMethod/Sources/AddTransferMethodPresenter.swift [hunk 17]
```diff
@@ -109,6 +117,7 @@ final class AddTransferMethodPresenter {
         }
         view.fieldValues().forEach { hyperwalletTransferMethod.setField(key: $0.name, value: $0.value) }
 
+        trackConfirmClick()
```
Comment (jsaini-hw) [inline TransferMethod/Sources/AddTransferMethodPresenter.swift]: move this to 1 line of this function

File: TransferMethod/Tests/AddTransferMethodPresenterTests.swift [hunk 18]
```diff
@@ -35,6 +40,11 @@ class AddTransferMethodPresenterTests: XCTestCase {
         XCTAssertTrue(mockView.isShowLoadingPerformed, "The showLoading should be performed")
         XCTAssertTrue(mockView.isHideLoadingPerformed, "The hideLoading should be performed")
         XCTAssertEqual(mockView.fieldGroups.count, 2, "The `response.getFields()` should be 2")
+        XCTAssertNotNil(HyperwalletInsights.shared, "")
+        XCTAssertNotNil(Insights.shared, "")
```
Comment (jsaini-hw) [inline TransferMethod/Tests/AddTransferMethodPresenterTests.swift]: Duplicate asserts

File: TransferMethod/Tests/AddTransferMethodPresenterTests.swift [hunk 19]
```diff
@@ -35,6 +40,11 @@ class AddTransferMethodPresenterTests: XCTestCase {
...
+        XCTAssertNotNil(Insights.shared, "Insights was not initialized")
```
Comment (jsaini-hw) [inline TransferMethod/Tests/AddTransferMethodPresenterTests.swift]: Let's not assert this here, you can remove it

File: TransferMethod/Tests/AddTransferMethodPresenterTests.swift [hunk 20]
```diff
@@ -1,5 +1,7 @@
+import Common
 import Hippolyte
 import HyperwalletSDK
+import Insights
```
Comment (jsaini-hw) [inline TransferMethod/Tests/AddTransferMethodPresenterTests.swift]: remove this import

File: HyperwalletUISDK.xcodeproj/project.pbxproj (hunk 8 above)
Comment (sarun-altmtrk) [reply HyperwalletUISDK.xcodeproj/project.pbxproj]: Due to change in latest refactored code, these fixes has been done in following PR: https://github.com/hyperwallet/hyperwallet-ios-ui-sdk/pull/167

File: TransferMethod/Tests/AddTransferMethodPresenterTests.swift (hunk 20 above)
Comment (sarun-altmtrk) [reply TransferMethod/Tests/AddTransferMethodPresenterTests.swift]: Due to change in latest refactored code, these fixes has been done in following PR: https://github.com/hyperwallet/hyperwallet-ios-ui-sdk/pull/167

**Comment by sarun-altmtrk** [comment]
Due to change in latest refactored code, these fixes has been done in following PR: https://github.com/hyperwallet/hyperwallet-ios-ui-sdk/pull/167
